### PR TITLE
Initial Deployment Docs

### PR DESCRIPTION
**Pull request by steven-exarcos**
A document covering the `infrastructure-documenation/deployment` topic superseding the legacy docs in https://github.com/eclipse-pass/main/tree/main/docs/infra.

### CODE REVIEW

**Comment by steven-exarcos** [comment]
@tsande16 and @cshannon2 This would be a starting baseline condensing of the pre-existing information in the category. 

https://github.com/eclipse-pass/main/pull/1008 subtracts the docs that feed into this one, or are obsolete in general.

**Comment by tsande16** [comment]
@steven-exarcos this looks like a good first start!

File: infrastructure-documenation/deployment/README.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+### GitHub Actions Workflow
+
+The "Publish: Release All" workflow automates the release process:
+
+1. Builds and tests components
+2. Publishes Java artifacts
+3. Builds and pushes Docker images
+4. Creates GitHub Releases
+
+For detailed configuration, refer to the `pass-complete-release.yml` file in the AWS-PASS-Deployment repository.
```
Comment (tsande16) [inline infrastructure-documenation/deployment/README.md:92]: Would be good to provide a link to this file, for the convenience of the user.

**Review by tsande16 — APPROVED** [review]
Content looks really good. Only 1 small suggestion. 

We've been using the following layout for our "flow" from the style guide: https://docs.google.com/document/d/11aCooQCNhEq34yG9mGuFynY4xMDRJtPRuOOou9LaCgU/edit?usp=sharing

Topic 
Topic Summary
Knowledge Needed / Skills Inventory
Technologies Utilized
Technical Deep Dive 
Next Step / Institution Configuration
Related Information

Although after reading this, it seems that the deployment guide could follow a different layout/structure. For now, I think this is a good flow and Chris and I can discuss later.

File: infrastructure-documenation/deployment/README.md (hunk 1 above)
Comment (steven-exarcos) [reply infrastructure-documenation/deployment/README.md:92]: @tsande16 The link would be to an internal repo that the public would not be able to see.  Should I include the link anyway.....or perhaps remove the mention?

Comment (tsande16) [reply infrastructure-documenation/deployment/README.md:92]: At least how it is now, the release workflows are in the `pass-main`: https://github.com/eclipse-pass/main/blob/de2018560410ffcbaab5ad3079d3c6cfa64aeeaa/.github/workflows/pass-complete-release.yml

The deployment to JHU PROD is in the `JHU-PASS-Deployment`. Perhaps there were discussions to migrate the release workflows over? I'm not aware of any at the moment, but then again I'm not in all the CI/CD discussions.

Comment (steven-exarcos) [reply infrastructure-documenation/deployment/README.md:92]: Thats a good point - there are actually *two* workflows and I was pointing to Deploy and calling it Release.

**Comment by steven-exarcos** [comment]
@tsande16 I renamed the sections to match the style guide where it made sense.   

I agree that for the middle technical sections, Knowledge Needed / Skills Inventory, Technologies Utilized, Technical Deep Dive, Next Step / Institution Configuration, they dont fit as smoothly when describing a process versus a spec. 

Also -- since I'm a new in eclipse org, I dont have permissions to merge yet.

File: infrastructure-documenation/deployment/README.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+Please refer to [github-cicd.md](./github-cicd.md) for further information on its useage.
+
+## Related Information
+
+- [PASS main repository](https://github.com/eclipse-pass/main)
+- [PASS Docker repository](https://github.com/eclipse-pass/pass-docker)
+
+For further assistance or questions, please open an issue in the [PASS main repository](https://github.com/eclipse-pass/main/issues) or find us in the [PASS Slack](https://eclipse-pass.slack.com)!
+```
```
Comment (tsande16) [inline infrastructure-documenation/deployment/README.md:90]: I think this creating a code block:

image

File: infrastructure-documenation/deployment/github-cicd.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+          docker push $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG
+
+```
+
+This workflow builds a Docker image and pushes it to Amazon ECR.
+
+## Related Information
+
+For more detailed information, refer to the [GitHub Actions documentation](https://docs.github.com/en/actions).
+```
```
Comment (tsande16) [inline infrastructure-documenation/deployment/github-cicd.md:144]: Looks like an extra code block here.

**Review by tsande16 — COMMENTED** [review]
Noticed one more thing. Looks like there is an added code block at the end of the files for the CI/CD and README?

File: infrastructure-documenation/deployment/README.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
+# Eclipse PASS Deployment Guide
+
+## Table of Contents
+1. [Summary](#summary)
+2. [Deployment Architecture](#deployment-architecture)
```
Comment (tsande16) [inline infrastructure-documenation/deployment/README.md:5]: With the changed headers some of these links are broken.

File: infrastructure-documenation/deployment/README.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+3. [AWS Infrastructure Components](#aws-infrastructure-components)
+4. [PASS Deployment Process](#pass-deployment-process)
+5. [PASS Release Process](#pass-release-process)
+6. [Related Information](#related-infomormation)
```
Comment (tsande16) [inline infrastructure-documenation/deployment/README.md:9]: link is broken here too.

File: infrastructure-documenation/deployment/roadmap.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
+# Eclipse PASS Cloud-Native Kubernetes Deployment Roadmap
+
+## Table of Contents
+1. [Introduction](#introduction)
+2. [Current State](#current-state)
+3. [Target Architecture](#target-architecture)
+4. [Roadmap Phases](#roadmap-phases)
+5. [Key Milestones](#key-milestones)
+6. [Challenges and Considerations](#challenges-and-considerations)
+7. [Timeline](#timeline)
```
Comment (tsande16) [inline infrastructure-documenation/deployment/roadmap.md:10]: I think the section header name changed here, or was removed.

**Review by tsande16 — COMMENTED** [review]
Looks like with the change of header names it broke some links.

**Pull request merged**